sandbox_dns_unix.go

```diff
@@ -184,6 +187,13 @@ func (sb *sandbox) setupDNS() error {
 			return err
 		}
 	} else {
+		// If embedded DNS server is enabled later for this contianer ipv4 loopback nameservers if
+		// present should be used. retain it before filtering out resolv.conf contents.
```

Comment:
I think handling the `--dns=127.0.0.1` in the Host NS make sense. But i think we should retain the filtering behavior for both default and user-defined networks.

With this change, we filter out the `127.0.0.1` from the docker0 default network , but retaining it for user-defined network and handling it in host namespace.

@mrjana wdyt ?